### PR TITLE
Issue#168: give friendlier error when port cant be converted to int

### LINKED ISSUE #168: Fix a error for invalid port
Now the exception is cryptic:
```
In [3]: URL('//https:www.twitter.com/Bard Tony')
---------------------------------------------------------------------------
ValueError                                Traceback (most recent call last)
<ipython-input-3-71922a9dd380> in <module>()
----> 1 URL('//https:www.twitter.com/Bard Tony')

~/.virtualenvs/aiohttp/lib/python3.6/site-packages/yarl/__init__.py in __init__(self, val, encoded, strict)
    168                                            val.password,
    169                                            host,
--> 170                                            val.port,
    171                                            encode=True)
    172             path = self._PATH_QUOTER(val[2])

/usr/lib/python3.6/urllib/parse.py in port(self)
    165         port = self._hostinfo[1]
    166         if port is not None:
--> 167             port = int(port, 10)
    168             if not ( 0 <= port <= 65535):
    169                 raise ValueError("Port out of range 0-65535")

ValueError: invalid literal for int() with base 10: 'www.twitter.com'

```

### PR DESCRIPTION
**Pull request by gyermolenko**
Fixes #168 

Few tests for bad port scenario are already present in `test_url_parsing.py`, they already cover current situation.

### CODE REVIEW

File: yarl/__init__.py [hunk 1]
```diff
@@ -164,10 +164,17 @@ def __new__(cls, val='', *, encoded=False, strict=None):
                 if host is None:
                     raise ValueError(
                         "Invalid URL: host is required for abolute urls.")
+
+                try:
+                    port = val.port
+                except ValueError:
+                    raise ValueError(
+                        "Invalid URL: port can't be converted to integer")
```
Comment (webknjaz) [inline yarl/__init__.py]: Maybe `raise ... from ...`?

Comment (asvetlov) [reply yarl/__init__.py]: Implicit exception context is good enough

**Comment by asvetlov** [comment]
thanks

**Comment by gyermolenko** [comment]
@webknjaz 
I found this issue from pyflakes repo https://github.com/PyCQA/pyflakes/issues/367

**Comment by asvetlov** [comment]
For every Python pre-release failures are expected.
We saw them for Python 3.5, 3.6 and 3.7.
Unsurprisingly 3.8 is not exceptional.

**Pull request merged**